### PR TITLE
0.20.0 Create session object so plugins can do advanced things

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Rivet/Functions/New-RivetSession.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
+
+function New-RivetSession
+{
+    <#
+    .SYNOPSIS
+    Creates a Rivet session object.
+
+    .DESCRIPTION
+    The `New-RivetSession` function creates a Rivet session. By default, the function will use the configuration from
+    the `rivet.json` in the current directory. To use a custom `rivet.json` file, pass its path to the `rivet.json` file
```
Comment (decoyjoe) [inline Rivet/Functions/New-RivetSession.ps1]: This sentence is awkward:
> To use a custom `rivet.json` file, pass its path to the `rivet.json` file to use for the session's configuration to the `ConfigurationPath` parameter.

File: Rivet/Functions/New-RivetSession.ps1 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    .EXAMPLE
+    New-RivetSession -ConfigurationPath '.\rivet.json'
+
+    Demonstrates how to create a Rivet session by passing the path to a `rivet.json` file to the `ConfigurationPath`
+    parameter.
+
+    .EXAMPLE
+    New-RivetSession -ConfigurationPath '.\rivet.json' -Database @('UseThisOne', 'AndThisOne')
+
+    Demonstrates how to create a session objerct that only operates on a specific database. IN this example, Rivet will
```
Comment (decoyjoe) [inline Rivet/Functions/New-RivetSession.ps1]: Typo: Casing on `IN`

File: Rivet/Rivet.psm1 [hunk 3]
```diff
@@ -177,6 +221,7 @@ Test-RivetTypeDataMember -TypeName 'Rivet.Scale' -MemberName 'Value'
     Join-Path -Path $rivetModuleRoot -ChildPath 'Functions'
     Join-Path -Path $rivetModuleRoot -ChildPath 'Functions\Columns'
     Join-Path -Path $rivetModuleRoot -ChildPath 'Functions\Operations'
+    Join-Path -Path $rivetModuleRoot -ChildPath 'Functions\Metadata'
```
Comment (decoyjoe) [inline Rivet/Rivet.psm1]: What's this `Functions\Metadata` directory?

File: global.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
     "sdk":  {
-                "version":  "3.1.201"
+                "version":  "6.0.408"
```
Comment (decoyjoe) [inline global.json]: Add a [`rollForward`](https://learn.microsoft.com/en-us/dotnet/core/tools/global-json) property?

**Review by decoyjoe — APPROVED** [review]

File: Rivet/Rivet.psm1 (hunk 3 above)
Comment (splatteredbits) [reply Rivet/Rivet.psm1]: I was going to start introducing metadata functions into Rivet (e.g. Get-ExtendedProperty, Get-Table, etc.) for use in plug-ins, but decided they fit better in Yodel than in Rivet, so removed them. Looks like I missed this spot.

**Pull request merged**